src/merge_right.rs

```diff
@@ -1,13 +1,65 @@
 use std::collections::{BTreeMap, BTreeSet, HashSet};
 
+use crate::config::{HttpVersion, Proxy, ScriptOptions};
+
 pub trait MergeRight {
     fn merge_right(self, other: Self) -> Self;
 }
 
-impl<A> MergeRight for Option<A> {
-    fn merge_right(mut self, other: Self) -> Self {
-        self = other.or(self);
-        self
+impl<A: MergeRight> MergeRight for Option<A> {
+    fn merge_right(self, other: Self) -> Self {
+        match (self, other) {
+            (Some(this), Some(that)) => Some(this.merge_right(that)),
+            (None, Some(that)) => Some(that),
+            (Some(this), None) => Some(this),
+            (None, None) => None,
+        }
+    }
+}
+
+pub trait Scalar {}
+
+impl Scalar for u64 {}
+
+impl Scalar for u32 {}
+
+impl Scalar for u16 {}
+
+impl Scalar for u8 {}
+
+impl Scalar for usize {}
+
+impl Scalar for i64 {}
+
+impl Scalar for i32 {}
+
+impl Scalar for i16 {}
+
+impl Scalar for i8 {}
+
+impl Scalar for f64 {}
+
+impl Scalar for f32 {}
+
+impl Scalar for bool {}
+
+impl Scalar for char {}
+
+impl Scalar for String {}
+
+impl Scalar for HttpVersion {}
```

Comment:
Create a separate file under src called `scalar`. Where all the scalars are implemented.

Reply:
We can even call it `Primitive` since graphQL scalars already exist.

Reply:
done file called primitives